src/jvmMain/kotlin/com/charleskorn/kaml/JvmYamlReading.kt

```diff
@@ -0,0 +1,61 @@
+/*
+
+   Copyright 2018-2023 Charles Korn.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       https://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+package com.charleskorn.kaml
+
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.serializer
+import okio.Source
+import okio.source
+import java.io.InputStream
+import java.nio.charset.Charset
+
+public fun <T> Yaml.decodeFromStream(
+    deserializer: DeserializationStrategy<T>,
+    source: InputStream,
+    charset: Charset = Charsets.UTF_8,
+): T {
+    return decodeFromSource(
+        deserializer,
+        source.source(),
+        charset,
+    )
+}
```

Comment:
Just my five cents, as a user of kaml: Unlike - as far as I know - kotlinx-serialization-json, kaml's API offers to decode from and to streams (java InputStream / okio Source). I think it is no surprise that kotlinx-serialization-json does not offer such an API, for currently - as far as I know -, there is no official Kotlin multiplatform library that offers a file access or even just a input/output stream API.

Without reading/writing to an input/output byte stream but only handling Strings, the official json serialization library does not have to deal with charsets at all.

While not even the official kotlinx serialization libraries offer this feature, it seems unnecessary to go out of one's way to support an API that is itself not supported by the libraries one builds upon (in this case, for input/output byte stream reading, okio).

There is the official [`kotlinx-io-core`](https://github.com/Kotlin/kotlinx-io) kmp library that should deal with byte streams (et al), but it is still in its experimental phase (and at least currently, depends on okio anyway).

In my opinion, as long as no official stable kmp library exists that offers reading and writing byte array streams, it would be best to go even one step further and actually only support serialization from and to Strings. (Now, whether or not okio could be considered official / de-facto standard, hm...)

Reply:
Kotlinx Serialization does support Java and Okio streams :)

* https://github.com/Kotlin/kotlinx.serialization/blob/v1.6.3/formats/json/jvmMain/src/kotlinx/serialization/json/JvmStreams.kt
* https://github.com/Kotlin/kotlinx.serialization/blob/v1.6.3/formats/json-okio/commonMain/src/kotlinx/serialization/json/okio/OkioStreams.kt

It doesn't support different charsets though.

Reply:
Oh!

Reply:
I think it's OK to drop support for non-UTF-8 charsets for now - if anyone needs anything else, they can convert it themselves or submit a PR to add support for it 🙂